src/macrofier/types.mjs

```diff
@@ -359,14 +394,27 @@ function getSchemaShape(schema = {}, module = {}, { templateDir = 'types', name
     return insertSchemaMacros(result, schema, module, theTitle, parent, property)
   }
   else if (schema.anyOf || schema.oneOf) {
-    // borrow anyOf logic, note that schema is a copy, so we're not breaking it.
-    if (!schema.anyOf) {
-      schema.anyOf = schema.oneOf
+    const template = getTemplate(path.join(templateDir, 'anyOfSchema' + suffix))
+    let shape
+    if (template) {
+      if (!template.includes('SKIP')) {
```

Comment:
SKIP is added to skip anyOf schema shape generation, since for CPP it is treating as string currently, so shape generation is not required. Updated template with comments